docs/book/src/reference/metrics.md

```diff
@@ -69,6 +69,61 @@ NAME                                         AGE
 monitor-controller-manager-metrics-monitor   2m8s
 ```
 
+### RBAC for Prometheus
+
+When using [prometheus-operator](https://github.com/prometheus-operator) for
+cluster monitoring, you should ensure that necessary RBAC roles and permissions
+were applied to the namespace where the manager is deployed. Otherwise,
+Prometheus server will not be able to access the ServiceMonitor that describes
+the set of targets to be monitored by Prometheus.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Do we need to add it here? is not that in the Prometheus Operator doc? 
Is not better only so that if they cange that we do not get this doc outdated? 

What about a note with "warning" (see that will render very nice on the docs)

```
<aside class="warning">
<h2>RBC required for Prometheus Operator have access to the metrics</h2>

If you have not Promethues installed on the cluster and you are using Promethues 
Operator then be aware that when using [prometheus-operator][rbac-promethues-operator] for
cluster monitoring, you should ensure the necessary RBAC roles and permissions. Otherwise,
Prometheus server will not be able to access the ServiceMonitor that describes
the set of targets to be monitored by Prometheus.

</aside>

Also, note that these are ClusterRoles so does not need to be applied on the ns of the operator.


```

Reply:
Hi @camilamacedo86 @joelanford, I think that we should provide this as an optional tool and provide an api call for it.
In the documentation we will explain when it can be used and also enable changing the namespace and additional details that may change. The namespace and service account name would have a default, but we should add to the tool flags to update them.

Reply:
@sradco is specific for Prometheus Operator and Prometheus Operator describes how to do that, see: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/rbac.md

Therefore I do not think that Kubebuilder should be responsible for providing and maintaining helpers over "how to configure Prometheus Operator itself". The RBAC proposed to allow Prometheus Operator to have access to the ServiceMonitor used by each operator. However, users might have it set up already or prefer as @joelanford described allow this permission to specific SA instead.  

We here could/should highlight the "pre-requirement" to check the metrics with Prometheus Operator since we are suggesting its usage, see the metrics docs:

> Install Prometheus and Prometheus Operator. We recommend using [kube-prometheus](https://github.com/coreos/kube-prometheus#installing) in production **if you don’t have your own monitoring system**

Also, note that we are saying in the doc as well: 

> To achieve that, you can create a clusterRoleBinding to bind the clusterRole to the service account that your Prometheus server uses. 

Therefore, we just need a warning note pointing out the users to ensure that **if** they have Prometheus installed via the Operator then they need to ensure that Prometheus Operator has the required permissions RBAC to access the ServiceMonitor in the namespace where the Operator will be deployed and should link their documentation as well. ( We should not doc how Prometheus Operator works or how to config it, it is also part of the Prometheus Operator domain. )



 

Because